xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/AddressUtils.java

```diff
@@ -39,27 +15,6 @@ private AddressUtils() {
    *
    * @return A Base58Check encoded XRPL address in Classic Address form.
    */
-  public Address deriveAddress(final PublicKey publicKey) {
-    Objects.requireNonNull(publicKey);
-    return AddressCodec.getInstance().encodeAccountId(computePublicKeyHash(publicKey.value()));
-  }
-
-  /**
-   * Compute the RIPEMD160 of the SHA256 of the given public key, which can be encoded to an XRPL address.
-   *
-   * @param publicKey The public key that should be hashed.
-   *
-   * @return An {@link UnsignedByteArray} containing the non-encoded XRPL address derived from the public key.
-   */
-  @SuppressWarnings("UnstableApiUsage")
-  private UnsignedByteArray computePublicKeyHash(final UnsignedByteArray publicKey) {
-    Objects.requireNonNull(publicKey);
+  Address deriveAddress(final PublicKey publicKey);
```

Comment:
What would you think about moving this method to `KeyPairService` and getting rid of this interface entirely. One could argue that deriving an address should be functionality of a `KeyPairService` because you're doing an operation/derivation from a key.

The real thing I am guarding against here is having 20 different interfaces that have a small functionality footprint. IMO, this is one of the big problems with xrpl4j v2 -- it can be really difficult to remember which service classes do what.

Reply:
I agree, although I think this should all be done when we condense the KeyPairService into the WalletFactory. I've added this to #170